### DISCUSSION TITLE
Apply transforms to handlers built via `toHandlers`

### DISCUSSION THREAD

**Discussion by michael-wolfenden** [mswjs/data#223] [Q&A]
Is is possible to apply addition transforms to handlers built via `toHandlers`, for example to introduce a delay to each handler.

**Comment by kettanaito** [comment]
Hey, @michael-wolfenden.

No, currently that is not possible. Can you tell me more about your use case? What is the ideal API in your opinion to customize `.toHandlers()` behavior?

**Reply by michael-wolfenden** [reply]
What if you could pass your own custom `response` method as shown in the example below:
https://mswjs.io/docs/api/response#custom-response-composition

**Reply by kettanaito** [reply]
You can define a delay that affects all generated handlers by creating a passthrough handler:

```js
const handlers = [
  rest.all('/user/*', async (req, res, ctx) => {
    // This will delay any requests to "/user" by 1000ms
    // and then perform them as-is, falling further to the
    // generated handlers below. 
    await new Promise(resolve => setTimeout(resolve, 1000))
    return req.passthrough()
  }),
  ...db.user.toHandlers('/user'),
]
```

**Reply by kettanaito** [reply]
I'm thinking that it'd more flexible to turn the intention around: have something like `.toTransformers()` that you can supply to any `res()` call:

```ts
rest.get('/route', (req, res, ctx) => {
  return res(...db.user.toTransformers())
})
```

The issue here is that DB would lose direct access to the request info like parameters, creating a surface for user errors and complicating the call signature since now you need to forward those parameters manually.

**Reply by michael-wolfenden** [reply]
Thanks @kettanaito, I'll give the passthrough handler a try
